src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1351,6 +1344,11 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
                 {
                     _HyperlinkHandler(_terminal->GetHyperlinkAtPosition(terminalPosition));
                 }
+                else if (_CanSendVTMouseInput())
+                {
+                    // GH#9396: hyper-link gets higher priority than mouse mode
+                    _TrySendMouseEvent(point);

```

Comment:
I'm a bit worried about the flow of code through these conditionals. This will change the block selection state, track click counts, etc. before dispatching a mouse event.

We might be overdue for refactoring the click dispatcher. . . but this seems a bit too entangled for my taste?

Reply:
@DHowett you are right as usual, currently the hyperlink handling was already entangled into the selection processing code. This forced me to push the VT processing there as well (as hyperlink should precede). 

I issued another commit, trying to decouple everything into:
1. try to handle hyperlink
2. else try to handle VT mouse
3. else try to update selection (left-click)
4. else try to handle paste (right-click)